internal/gitShim/containerRegistryAuth.go

```diff
@@ -48,12 +49,20 @@ func CreateContainerRegistrySecret(obj *ContainerRegistryAuth) (string, error) {
 			usernamePasswordStringB64,
 		)
 
+		data := map[string][]byte{"config.json": []byte(dockerConfigString)}
 		argoDeployTokenSecret := &v1.Secret{
 			ObjectMeta: metav1.ObjectMeta{Name: secretName, Namespace: "argo"},
-			Data:       map[string][]byte{"config.json": []byte(dockerConfigString)},
+			Data:       data,
 			Type:       "Opaque",
 		}
-		if err := k8s.CreateSecretV2(obj.Clientset, argoDeployTokenSecret); err != nil {
+		err := k8s.CreateSecretV2(obj.Clientset, argoDeployTokenSecret)
+		if errors.IsAlreadyExists(err) {
+			if err := k8s.UpdateSecretV2(obj.Clientset, "argo", secretName, data); err != nil {
+				return "", fmt.Errorf("error while updating secret for GitHub container registry auth: %w", err)
+			}
+		}
+
+		if err != nil {
```

Comment:
You might wanna check too for...

```suggestion
		if err != nil && !errors.IsAlreadyExists(err) {
```

The `errors.IsAlreadyExists()` has a tree: if it already exists and you updated the flow successfully, `err` will still be true to `errors.IsAlreadyExists()` when you hit this line. The `err` on `UpdateSecretV2` is scoped to the `if`.

Reply:
Yup, Good catch. for some reason I removed that last night and don't recall why